database/src/migrations/20240308110000_survey_progress.ts

```diff
@@ -0,0 +1,93 @@
+      survey_progress_id                            integer            GENERATED ALWAYS AS IDENTITY (START WITH 1 INCREMENT BY 1),
+      name                                          varchar(32)        NOT NULL,
+      description                                   varchar(128),
+      record_end_date                               timestamptz(6),
+      create_date                                   timestamptz(6)     DEFAULT now() NOT NULL,
+      create_user                                   integer            NOT NULL,
+      update_date                                   timestamptz(6),
+      update_user                                   integer,
+      revision_count                                integer            DEFAULT 0 NOT NULL,
+      CONSTRAINT survey_progress_id_pk PRIMARY KEY (survey_progress_id)
+    );
+
+    COMMENT ON TABLE survey_progress IS 'This table is intended to store options that users can select for their survey status.';
+    COMMENT ON COLUMN survey_progress.survey_progress_id IS 'Primary key for survey_progress.';
+    COMMENT ON COLUMN survey_progress.name IS 'Name of the survey progress option.';
+    COMMENT ON COLUMN survey_progress.description IS 'Description of the survey progress option.';
+    COMMENT ON COLUMN survey_progress.record_end_date IS 'End date of the survey progress option.';
+    COMMENT ON COLUMN survey_progress.create_date IS 'The datetime the record was created.';
+    COMMENT ON COLUMN survey_progress.create_user IS 'The id of the user who created the record as identified in the system user table.';
+    COMMENT ON COLUMN survey_progress.update_date IS 'The datetime the record was updated.';
+    COMMENT ON COLUMN survey_progress.update_user IS 'The id of the user who updated the record as identified in the system user table.';
+    COMMENT ON COLUMN survey_progress.revision_count IS 'Revision count used for concurrency control.';
+
+    ----------------------------------------------------------------------------------------
+    -- Add triggers for user data
+    ----------------------------------------------------------------------------------------
+    CREATE TRIGGER audit_survey_progress BEFORE INSERT OR UPDATE OR DELETE ON biohub.survey_progress FOR EACH ROW EXECUTE PROCEDURE tr_audit_trigger();
+    CREATE TRIGGER journal_survey_progress AFTER INSERT OR UPDATE OR DELETE ON biohub.survey_progress FOR EACH ROW EXECUTE PROCEDURE tr_journal_trigger();
+
+    ----------------------------------------------------------------------------------------
+    -- Modify survey table to include survey_progress
+    ----------------------------------------------------------------------------------------
+    ALTER TABLE survey ADD COLUMN progress_id INTEGER NOT NULL;
+    COMMENT ON COLUMN survey.progress_id IS 'Foreign key referencing the progress value.';
+    ALTER TABLE survey ADD CONSTRAINT survey_progress_fk FOREIGN KEY (progress_id) REFERENCES survey_progress(survey_progress_id);
```

Comment:
Unlike primary keys, foreign keys in postgres dont come with an index, so you have to add one manually (just to improve performance).

Should be examples in the other recent migrations

Reply:
Your sample block migration includes them correctly

Reply:
Added index